tests/core/test_addresses_correctly_assigned.py

```diff
@@ -14,29 +12,35 @@
 ]
 
 
+@pytest.fixture
+def get_payment(get_order_and_payment):
+    def _create_payment_with_info_data():
+        info_data = {
+            'currency_type': 'ETH',
+            'time': int(time.time()),
+            'amount': 1
+        }
+        _, payment = get_order_and_payment(info_data=info_data)
+        return payment
+    return _create_payment_with_info_data
+
+
 @pytest.mark.django_db
-def test_addresses_correctly_assigned(event, provider, get_order_and_payment):
+def test_addresses_correctly_assigned(event, provider, get_payment):
     provider.settings.set('ETH_RATE', '1.0')
 
-    for address in hex_addresses:
-        WalletAddress.objects.create(event=event, hex_address=address)
+    assert WalletAddress.objects.filter(order_payment__isnull=True).count() == 0
+
+    WalletAddress.objects.bulk_create(
+        [WalletAddress(event=event, hex_address=address) for address in hex_addresses])
 
     assert WalletAddress.objects.filter(order_payment__isnull=True).count() == len(hex_addresses)
 
     used_addresses = set()
 
     for _ in range(len(hex_addresses)):
-        factory = RequestFactory()
-        request = factory.get('/checkout')
-
-        info_data = {
-            'currency_type': 'ETH',
-            'time': int(time.time()),
-            'amount': 1
-        }
-        _, payment = get_order_and_payment(info_data=info_data)
-
-        response_page = provider.payment_pending_render(request, payment)
+        payment = get_payment()
+        response_page = provider.payment_pending_render(None, payment)
```

Comment:
I'm :-1: on passing in a `None` where a different argument type is expected, even if the argument isn't being used.  Better to either go back to what you were doing before (even though the argument isn't used) or to fully remove it from the API.

Reply:
I see. I will revert because removing it from API is trouble since this function is part of pretix interface.